Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -183,10 +184,43 @@ namespace ROS2
         // Build up a list of all entities created as a part of processing the file.
         AZStd::vector<AZ::EntityId> createdEntities;
         AZStd::unordered_map<const sdf::Link*, AzToolsFramework::Prefab::PrefabEntityResult> createdLinks;
+        AZStd::unordered_map<const sdf::Model*, AzToolsFramework::Prefab::PrefabEntityResult> createdModels;
         AZStd::unordered_map<AZStd::string, const sdf::Link*> links;
-        for ([[maybe_unused]] const auto& [fullLinkName, linkPtr, _] : linksMapper.m_links)
+        for ([[maybe_unused]] const auto& [fullLinkName, linkPtr, attachedModel] : linksMapper.m_links)
         {
-            createdLinks[linkPtr] = AddEntitiesForLink(linkPtr, AZ::EntityId{}, createdEntities);
+            // Create entities for the model containing the link
+            AZ::EntityId modelEntityId;
+            if (attachedModel != nullptr)
+            {
+                if (auto modelIt = createdModels.find(attachedModel); modelIt != createdModels.end())
+                {
+                    if (modelIt->second.IsSuccess())
+                    {
+                        modelEntityId = modelIt->second.GetValue();
+                    }
+                }
+                else
+                {
+                    if (AzToolsFramework::Prefab::PrefabEntityResult createModelEntityResult = CreateEntityForModel(*attachedModel);
+                        createModelEntityResult)
+                    {
+                        modelEntityId = createModelEntityResult.GetValue();
+                        // Add the model entity to the created entity list
+                        // so that it gets added to the prefab
+                        createdEntities.emplace_back(modelEntityId);
+
+                        std::string modelName = attachedModel->Name();
+                        AZStd::string azModelName(modelName.c_str(), modelName.size());
```

Comment:
```suggestion
                        AZStd::string azModelName(modelName.data(), modelName.size());
```

Reply:
There is no differences between `.data()` and `.c_str()` and neither one is better in this case, plus the C++ Best Practices Guide recommends using `.c_str()` over `.data()` when accessing the string internal buffer: https://github.com/o3de/sig-core/blob/main/governance/C%2B%2B-Best-Practices-Guide.md#string-buffer-access

Reply:
IMO the Best Practice doesn't apply in this case, since you're explicitly intentionally ignoring the null termination and creating a new string without a null terminator. By using c_str(), it's guaranteeing that a null-terminated string is getting generated if it doesn't already exist. data() allows you to create the new string without requiring the null terminator.

If you want to use c_str() and preserve the terminator, then shouldn't this use modelName.size() + 1?

Reply:
`c_str()` and `data()` are literally the same implementation [per the C++ standard](https://en.cppreference.com/w/cpp/string/basic_string/data) post C++11
> The returned array is null-terminated, that is, data() and [c_str()](https://en.cppreference.com/w/cpp/string/basic_string/c_str) perform the same function.